CONTRIBUTING.md

```diff
@@ -0,0 +1,165 @@
+
+So, the first step is to fork the [Phootwork repository](https://github.com/phootwork/phootwork) on Github and clone it
+on your local machine. If you have any doubt, please read https://help.github.com/en/articles/fork-a-repo .
+
+Then install the dependencies:
+```bash
+composer install
+```
+
+## Understanding the directory structure
+
+After cloning and installing the dependencies, you should have a directory structure like the following:
+```.
+   ├── bin
+   ├── src
+   │   ├── collection
+   │   ├── file
+   │   │   └── exception
+   │   ├── json
+   │   ├── lang
+   │   │   └── text
+   │   ├── tokenizer
+   │   └── xml
+   │       └── exception
+   ├── tests
+   │   ├── collection
+   │   │   └── fixtures
+   │   ├── file
+   │   ├── json
+   │   ├── lang
+   │   │   ├── fixtures
+   │   │   └── text
+   │   ├── tokenizer
+   │   │   └── fixtures
+   │   │       └── samples
+   │   └── xml
+   │       └── fixtures
+   └── vendor
+```
+
+Into the directory `bin` we have two scripts: `auto_split.sh` splits the whole Phootwork repository into the single
+library packages and `build_api.sh` generates the api documentation. You usually don't care about them, because they are
+run by our continuous integration server [Travis ci](https://travis-ci.org).
+
+As usual, into the directory `vendor` you can find all the libraries Phootwork depends on, both for runtime and development.
+
+The directory `src` contains the source code of our libraries and `tests` contains the relative tests.
+
+Both `src` and `tests` have sub-folders which correspond to the single libraries. E.g. you can find the source code of
+the library https://github.com/phootwork/lang into `src/lang` folder and the relative tests into `tests/lang` directory. 
+ 
+## Running the test suite
+
+While developing, the test part is very important: if you apply a patch to the existing code, the test suite must run without
+errors or failures and if you add a new functionality, no one will consider it without tests.
+
+Our test tool is [PhpUnit](https://phpunit.de/) and you can launch the test suite by running the following
+command from the phootwork root directory:
+
+```bash
+vendor\bin\phpunit
+```
+
+Each single library has its own test suite and you can launch it separately. I.e. if you want to run the `phootwork/lang`
+test suite:
+
+```bash
+vendor/bin/phpunit --testsuite lang
+```
+
+The last command can be useful to speed up the tests, if your contribution involves only one library.
+
+## Static analysis tool
+
+To prevent as many bugs as possible, we use a static analysis tool called [Psalm](https://psalm.dev/).
+To launch it, run the following command from the root directory of Phootwork project:
+
+```bash
+vendor\bin\psalm
+```
+
+After its analysis, Psalm outputs errors and issues with its suggestions on how to fix them. Errors are more important
+and generally more dangerous than issues, anyway you should fix both.
+
+## Coding standard
```

Comment:
We can perhaps move this into a separate repo (for documentation purpose only), e.g. `phootwork/codestyle`? Dunno if this is too much, would maybe only be a readme, but can also contain files for various IDE to help assist their formatting (stuff I put into the cs-fixer repo for eclipse for example).

Reply:
I removed this paragraph and just explained how to use the commands. I linked https://github.com/phootwork/php-cs-fixer-config for insights.